src/App.tsx

```diff
@@ -1,24 +1,215 @@
+    updateTodo(id, !completed).then(() => {
+      setIsUpdating(prevState => !prevState);
+    }).catch(() => setError('Unable to update a todo'));
+  };
+
+  const handleAllComplete = () => {
+    const allInCompleted = filteredTodos.filter(todo => (
+      todo.completed === false
+    ));
+
+    if (allInCompleted.length === 0) {
+      filteredTodos.forEach(todo => {
+        updateTodo(todo.id, false).then(() => {
+          setIsUpdating(prevState => !prevState);
+        }).catch(() => setError('Unable to update a todo'));
+      });
+    }
+
+    allInCompleted.forEach(todo => {
+      updateTodo(todo.id, !todo.completed).then(() => {
+        setIsUpdating(prevState => !prevState);
+      }).catch(() => setError('Unable to update a todo'));
+    });
+  };
+
+  const handleEditTitle = (id: number, text: string) => {
+    updateTodo(id, text).then(() => {
+      setIsUpdating(prevState => !prevState);
+    }).catch(() => setError('Unable to update a todo'));
+  };
+
+  const todoElements = filteredTodos.map(todo => (
+    <Todo
+      todo={todo}
+      temp={!!deleteIds.includes(todo.id)}
+      handleRemoveTodo={handleRemoveTodo}
+      key={todo.id}
+      handleComplete={handleComplete}
+      handleEditTitle={handleEditTitle}
+    />
+  ));
+  const countNotCompletedtodos = todos.filter(todo => (
+    todo.completed === false
+  )).length;
+
+  const completedTodos = todos.filter(todo => (
+    todo.completed === true
+  ));
+
+  const handleRemoveCompletedTodos = () => {
+    completedTodos.forEach(completedTodo => {
+      removeTodo(completedTodo.id).then(() => {
+        setTodos(todos.filter((todo) => todo.id !== completedTodo.id));
+        setDeleteIds(prevState => ([
+          ...prevState,
+          completedTodo.id,
+        ]));
+      }).catch(() => setError('Unable to delete a todo'));
+    });
+  };
+
+  const activeTodos = todos.find(todo => todo.completed);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {activeTodos && (
+            <button
+              type="button"
+              className="todoapp__toggle-all active"
+              onClick={handleAllComplete}
+            />
+          )}
+
+          <form onSubmit={handleAddTodo}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={addValue}
+              onChange={(event) => setAddValue(event.target.value)}
```

Comment:
It would be more clear this way
```suggestion
              onChange={(event) => setInputValue(event.target.value)}
```